tests/test.sh

```diff
@@ -26,14 +26,13 @@ else
 fi
 
 echo "Running all tests for $tag"
-find "$test_dir" -mindepth 1 -maxdepth 1 -type d | sed -E -e 's/^\.\///' > tests_to_run
-while read  -r d; do
+
+for d in $(find "$test_dir" -mindepth 1 -maxdepth 1 -type d | sed -E -e 's/^\.\///'); do
```

Comment:
I did this improvement too as part of this commit - avoids leaving tmp files in the file system after a run.